repo/manifest/manifest_manager_test.go

```diff
@@ -485,6 +488,108 @@ func TestManifestAutoCompactionWithReadOnly(t *testing.T) {
 	mgr, err = NewManager(ctx, bm, ManagerOptions{}, nil)
 	require.NoError(t, err, "getting other instance of manifest manager")
 
-	_, err = mgr.Find(ctx, map[string]string{"color": "red"})
+	entries, err := mgr.Find(ctx, map[string]string{"color": "red"})
 	require.NoError(t, err, "forcing reload of manifest manager")
+	assert.Len(t, entries, 100)
+}
+
+// BenchmarkConcurrentCompactionLoading benchmarks the runtime of loading
```

Comment:
I can remove this benchmark if folks would prefer not to keep benchmarks in the repo. Part of why it was committed was so I could cherry-pick it onto a branch with the original implementation for comparison